openjdk/mmtkMutator.hpp

```diff
@@ -38,9 +39,18 @@ struct LargeObjectAllocator {
     RustDynPtr plan;
 };
 
+struct MallocAllocator {
+    void* tls;
+    void* cursor;
+    void* limit;
+    RustDynPtr space;
+    void* plan;
+};
```

Comment:
This is inconsist with the MallocAllocator definition in Rust:

```rust
#[repr(C)]
pub struct MallocAllocator<VM: VMBinding> {
    pub tls: OpaquePointer,
    space: Option<&'static dyn Space<VM>>,
    plan: &'static dyn Plan<VM = VM>,
}
```

So here we only need three c++ struct fields, the first is a normal pointer, and the other two are rust dyn pointers:

```c++
struct MallocAllocator {
    void* tls;
    RustDynPtr space;
    RustDynPtr plan;
};
```